collie/metrics.py

```diff
@@ -49,129 +53,159 @@ def _get_user_item_pairs(user_ids: Union[np.array, torch.tensor],
         np.array([0, 1, 2, 3, 0, 1, 2, 3, 0, 1, 2, 3])
 
     """
-    # Added because sometimes we call this function with ``n_items`` as ``np.int64`` type which
-    # breaks ``repeat_interleave``.
-    if isinstance(n_items, np.int64):
-        n_items = n_items.item()
+    try:
+        negative_sample_type = model.train_loader.negative_sample_type
+    except AttributeError:  # explicit models do not have ``negative_sample_type``
+        negative_sample_type = 'item'
```

Comment:
Same comment here - can we just pass in `negative_sample_type` here to avoid re-computing this each time?